clients/src/test/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumerTest.java

```diff
@@ -157,9 +161,39 @@ public void testCommitted() {
     }
 
     @Test
-    public void testUnimplementedException() {
+    public void testAssign() {
+        consumer = spy(newConsumer(time, new StringDeserializer(), new StringDeserializer()));
+        consumer.assign(singleton(new TopicPartition("foo", 3)));
+        assertTrue(consumer.subscription().isEmpty());
+        assertTrue(consumer.assignment().isEmpty());
```

Comment:
Shouldn't it be expected to have the foo-3 assignment here? (assigned from user in the subscriptionState as a result of the previous consumer.assign)

Reply:
😓 - I guess the test never passed.